modules/caddyhttp/caddyhttp.go

```diff
@@ -413,6 +414,20 @@ type MiddlewareHandler interface {
 // emptyHandler is used as a no-op handler.
 var emptyHandler Handler = HandlerFunc(func(http.ResponseWriter, *http.Request) error { return nil })
 
+// An implicit suffix middleware that, if reached, sets the StatusCode to the
+// error stored in the ErrorCtxKey. This is to prevent situations where the
+// Error chain does not actually handle the error (for instance, it matches only
+// on some errors). See #3053
+var errorEmptyHandler Handler = HandlerFunc(func(w http.ResponseWriter, r *http.Request) error {
+	http_error := r.Context().Value(ErrorCtxKey)
```

Comment:
```suggestion
	httpError := r.Context().Value(ErrorCtxKey)
```

(Go doesn't conventionally use snake_case)